frontend/templates/troubleshooting/components/Causes.tsx

```diff
@@ -0,0 +1,267 @@
+import {
+   Box,
+   Heading,
+   HeadingProps,
+   Link,
+   Square,
+   useToken,
+} from '@chakra-ui/react';
+import { faList, faCircleNodes } from '@fortawesome/pro-solid-svg-icons';
+import { FaIcon } from '@ifixit/icons';
+import { useRef } from 'react';
+import { useScrollToActiveEffect } from '../toc';
+import {
+   TOCRecord,
+   useTOCBufferPxScrollOnClick,
+   useTOCContext,
+} from '../tocContext';
+
+function isIntroData(data: TOCEntry): data is IntroData & TOCRecord {
+   return data.type === 'Introduction';
+}
+function isCauseData(data: TOCEntry): data is SolutionData & TOCRecord {
+   return data.type === 'Solution';
+}
+
+function isConclusionData(data: TOCEntry): data is ConclusionData & TOCRecord {
+   return data.type === 'Conclusion';
+}
+
+export function Causes(
+   props: React.ComponentProps<typeof Box>,
```

Comment:
Chakra has a nice alias:

```suggestion
   props: BoxProps,
```

Reply:
Looks like we need this as it was. Good looking out though